gems/redis/4.2/redis.rbs

```diff
@@ -198,4 +198,8 @@ class Redis
   # @return [String] `PONG`
   def ping: (?String? message) -> String
 
+  def zcard: (String|Symbol key) -> Integer
+  def zadd: (String|Symbol key, *untyped args, ?nx: bool, ?xx: bool, ?lt: bool, ?gt: bool, ?ch: bool, ?incr: bool) -> [bool, Integer, Float]
```

Comment:
XX and NX look to be mutually exclusive, according to [ZADD](https://redis.io/commands/zadd/). This is tested in redis-rb [here](https://github.com/redis/redis-rb/blob/v4.2.5/test/lint/sorted_sets.rb#L46). In my opinion, it would be better if we can understand whether we use incompatible options at the same time before running tests.

By the way, `:lt` and `:gt` are not supported in v4.2; they were introduced in v4.5.0.

Also, [the tests for sorted sets](https://github.com/redis/redis-rb/blob/v4.2.5/test/lint/sorted_sets.rb) explains the return type of `#zadd` can be changed based on its arguments. In v4.2, `#zadd` possibly returns `bool`, `Integer`, or `Float` from the test assertions:

* https://github.com/redis/redis-rb/blob/v4.2.5/test/lint/sorted_sets.rb#L33
* https://github.com/redis/redis-rb/blob/v4.2.5/test/lint/sorted_sets.rb#L54
* https://github.com/redis/redis-rb/blob/v4.2.5/test/lint/sorted_sets.rb#L102

Ideally, we should declare the method signature for each condition (e.g., if `:incr` is passed, the return type will be `Float`), but it would be challenging to follow all cases (at least to me).

Now, I want to suggest the following code, following the above. What do you think?

```suggestion
  def zadd: (_ToS key, *untyped args, ?nx: bool, ?ch: bool, ?incr: bool) -> bool | Integer | Float
          | (_ToS key, *untyped args, ?xx: bool, ?ch: bool, ?incr: bool) -> bool | Integer | Float
```

Reply:
> By the way, :lt and :gt are not supported in v4.2; they were introduced in v4.5.0.

This was my lack of attention. Thanks for comments and suggestion!

> Ideally, we should declare the method signature for each condition (e.g., if :incr is passed, the return type will be Float), but it would be challenging to follow all cases (at least to me).

I agree that there is a benefit to precise definition, but I find it difficult to maintain perfection. (PR for further improvement is welcome)
I think your suggestion is a good balance for me.